main.go

```diff
@@ -111,6 +124,11 @@ func getCSS(rw http.ResponseWriter, req *http.Request, args map[string]string) {
 	}
 }
 
+func init() {
+	flag.Var(&attrs, "a", "period separated list of attrs to be assigned to nodes (one per node, in order) e.g. -a=ssd.ssd.ssd.hdd")
+	flag.Var(&localities, "l", "period separated list of localities to be assigned to nodes (one per node, in order) e.g. -l=l1=a,l2=a.l1=a,l2=b.l1=b,l1=a")
```

Comment:
Using period as a separator is unique. I think the `flag` package allows a flag to be specified multiple times and have each call add a new entry. Something like `-l l1=a,l2=a -l l1=a,l2=b -l l1=b,l2=a`. The change to `stringSlice.Set` to support this is trivial.

Also, it might make sense to make it explicit which node id the locality and attribute apply to. That is, `-l <nodeID>:<locality>`.